2.0/s2i/bin/run

```diff
@@ -7,9 +7,21 @@ function is_puma_installed() {
 
 set -e
 
+function check_number() {
+  if [[ ! "$2" =~ ^[0-9]+$ ]]; then
+    echo "$1 needs to be a non-negative number"
+    exit 1
+  fi
+}
+check_number PUMA_WORKERS     "${PUMA_WORKERS:-0}"
+check_number PUMA_MIN_THREADS "${PUMA_MIN_THREADS:-0}"
+check_number PUMA_MAX_THREADS "${PUMA_MAX_THREADS:-0}"
```

Comment:
Wouldn't it be better/easier to do this in Ruby?

I wouldn't expect Bash to have anything to do with this vars...

We simply launch Puma, it runs its puma.cfg script, and if the script fails the container/pod fails, etc...
If the env vars are present and valid, use them, if invalid, error, if not present, use defaults.